api/custom_auth/oauth/serializers.py

```diff
@@ -25,14 +25,25 @@ class Meta:
         abstract = True
 
     def create(self, validated_data):
-        user = self._get_user()
+        user_info = self.get_user_info()
+        if settings.AUTH_CONTROLLER_INSTALLED:
```

Comment:
Can we not call authenticate here as we did in the saml implementation?

Reply:
We can, but initially I wanted to avoid doing that since that's sort of not idiomatic - but I guess it's not so bad.

Reply:
I have moved it to use `authenticate`

Reply:
Yeah, ok, I see what you mean. It is a little strange calling authenticate in these serializer methods. I wonder if there's a middle ground where we add e.g. a static method on the `DomainAuthMethods` class and then call that in each of the serializers (and in the new BackEnd class in the auth module?)?

Reply:
I think that would be leaky abstraction, no? I think `DomainAuthMethods` is part of implementation detail and user does not need to know anything about it in order to interact with the library 

```
from auth_controller.controller import is_authentication_method_valid
is_authentication_method_valid(request, email)
```
Does that make sense?

Reply:
Yeah, I think you're right. But now I think I prefer the original version that you had 😄 so yeah, what you have in the comment here is better I think. Sorry!

Reply:
No prob, it was just a git reset away